### PR TITLE
[Merged by Bors] - Use latest tags for nethermind and geth in the execution engine integration test

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed

Currently the execution-engine-integration test uses latest master for nethermind and geth, and right now the test fails using the latest unreleased commits.

## Proposed Changes
Fix the nethermind and geth revisions the test uses to the latest tag in each repo. This way we are not continuously testing unreleased code, which might even get reverted, and reduce the failures only to releases in each one.
Also improve error handling of the commands used to manage the git repos.

## Additional Info
na

### CODE REVIEW

File: testing/execution_engine_integration/src/build_utils.rs [hunk 1]
```diff
@@ -15,51 +15,100 @@ pub fn prepare_dir() -> PathBuf {
...
-    Command::new("git")
-        .arg("pull")
-        .current_dir(repo_dir)
-        .output()
-        .unwrap_or_else(|_| panic!("failed to update branch at {:?}/{}", repo_dir, branch_name))
-        .status
-        .success()
+/// Gets the last annotated tag of the given repo.
+pub fn get_latest_release(repo_dir: &Path) -> Result<String, String> {
+    // In the directory was already present it is possible we don't have the most recent tags.
```
Comment (michaelsproul) [inline testing/execution_engine_integration/src/build_utils.rs]:
```suggestion
    // If the directory was already present it is possible we don't have the most recent tags.
```

File: testing/execution_engine_integration/src/geth.rs [hunk 2]
```diff
@@ -23,20 +22,17 @@ pub fn build(execution_clients_dir: &Path) {
...
+        build_utils::clone_repo(execution_clients_dir, GETH_REPO_URL).unwrap();
     }
 
-    // Checkout the correct branch
-    assert!(build_utils::checkout_branch(&repo_dir, GETH_BRANCH));
-
-    // Update the branch
-    assert!(build_utils::update_branch(&repo_dir, GETH_BRANCH));
+    // Get the latest tag on the branch
+    let last_release = build_utils::get_latest_release(&repo_dir).unwrap();
```
Comment (michaelsproul) [inline testing/execution_engine_integration/src/geth.rs]: I think this is slightly awkward in the case where `repo_dir` already existed and was checked out to a prior release. The `git describe` will describe the `HEAD` so it will just return the same tag back, and the repo will be stuck until the user manually intervenes. I don't think this is a problem for CI, however.

I guess the `update_branch` function is meant to address this, but it's unused in the current code. Alternatively we could do `git describe origin/{branch_name}` (after the `git fetch`), which avoids the problems of statefullness.

I'm inclined to merge this PR now as-is and we can come back for these changes if others agree?

**Review by michaelsproul — APPROVED** [review]
Approving and merging to unblock CI, thanks Diva!

**Comment by divagant-martian** [comment]
bors r-

I'm cancelling to address what you've raised

File: testing/execution_engine_integration/src/geth.rs (hunk 2 above)
Comment (divagant-martian) [reply testing/execution_engine_integration/src/geth.rs]: So I tried the describe approach and didn't work. I'll just do checkout branch, pull, fetch tags and then describe. What do you think?

Comment (michaelsproul) [reply testing/execution_engine_integration/src/geth.rs]: I think that's fine! I'm surprised the describe approach doesn't work though

Comment (divagant-martian) [reply testing/execution_engine_integration/src/geth.rs]: oh I think it might be because I testing against my lighthouse repo and I've never updated my remote master, let me try again

Comment (divagant-martian) [reply testing/execution_engine_integration/src/geth.rs]: yep it works

**Comment by michaelsproul** [comment]
It seems the latest Geth tag has some incompatibility, the failure doesn't seem related to your latest change

**Comment by divagant-martian** [comment]
the selected tag was not the right one, now should work

**Comment by divagant-martian** [comment]
bors retry
